File: tests/fixtures/languages/fsharp/sample.fs
namespace MyCorp

open System
open System.IO

module Geometry =

    type Point = { X: int; Y: int }

    type Color =
        | Red
        | Green of int
        | Blue of name: string

    let origin = { X = 0; Y = 0 }

    let rec distance (a: Point) (b: Point) : float =
        sqrt (float ((a.X - b.X) * (a.X - b.X) + (a.Y - b.Y) * (a.Y - b.Y)))

    type FooBuilder() =
        member _.Return(x) = [x]

    let foo = FooBuilder()

File: tests/fixtures/languages/typescript/sample.ts
import { useState } from "react";
import type { ReactNode } from "react";
import * as fs from "fs";

export type Box<T> = { value: T };
export interface Container<T> {
    items: T[];
}
export enum Color { Red, Green, Blue }

namespace Geometry {
    export class Point { constructor(public x: number, public y: number) {} }
    export function distance(a: Point, b: Point): number { return 0; }
}

export class Widget {
    items: Color[] = [];
    render(): ReactNode { return null; }
}

export default function () { return new Widget(); }
